Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,362 @@
+    }
+
+    super.tearDown()
+  }
+
+  func testFileByFileName() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.fileByFilename = "bar1.proto"
+      }
+    )
+    serviceReflectionInfo.requestStream.finish()
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(
+        serializedData: (message?.fileDescriptorResponse
+          .fileDescriptorProto[0])!
+      )
+    XCTAssertEqual(receivedFileDescriptorProto.name, "bar1.proto")
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.count,
+      1
+    )
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.first!.method.first!.name,
+      "testMethod1"
+    )
+    XCTAssertEqual(message?.fileDescriptorResponse.fileDescriptorProto.count, 4)
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+
+    try await serviceReflectionInfo.requestStream.send(
+      .with {
+        $0.host = "127.0.0.1"
+        $0.listServices = "services"
+      }
+    )
+
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }
+    XCTAssertEqual(receivedServices?.count, 4)
+
+    let servicesNames = self.getServicesNamesFromProtos(protos: self.makeProtosWithDependencies())
+    for serviceName in receivedServices! {
+      XCTAssertTrue(servicesNames.contains(serviceName))
+    }
+  }
+
+  func testReflectionServiceDataFileDescriptorDataByFilename() throws {
+    var protos = self.makeProtosWithDependencies()
+    let registry = try ReflectionServiceData(fileDescriptor: protos)
+
+    let registryFileDescriptorData = registry.fileDescriptorDataByFilename
+
+    for (fileName, protoData) in registryFileDescriptorData {
+      let serializedFiledescriptorData = protoData.serializedFileDescriptorProto
+      let dependecies = protoData.dependencyNames
+
+      let originalIndex = protos.firstIndex(where: { $0.name == fileName })
+      XCTAssertNotNil(originalIndex)
+
+      let originalProto = protos[originalIndex!]
+      XCTAssertEqual(originalProto.name, fileName)
+      XCTAssertEqual(try originalProto.serializedData(), serializedFiledescriptorData)
+      XCTAssertEqual(originalProto.dependency, dependecies)
+
+      protos.remove(at: originalIndex!)
```

Comment:
Rather than banging `originalIndex`, arrange the code so that you don't have to bang it:

```swift
guard let index = protos.firstIndex(where: { $0.name == fileName }) else {
  return XCTFail("....")
}

let proto = protos[index]
protos.remove(at: index)

// test things...

```